x-pack/packages/ai-infra/product-doc-artifact-builder/src/tasks/process_documents.ts

```diff
@@ -56,12 +56,21 @@ const processDocument = (document: ExtractedDocument) => {
     })
     // remove edit links
     .replaceAll(/\[\s*edit\s*\]\(\s*[^)]+\s*\)/g, '')
-    // remove empty links
-    .replaceAll('[]()', '')
+    // remove empty and image links
+    .replaceAll(/\[\]\(\s*[^)]+\s*\)/g, '')
     // limit to 2 consecutive carriage return
     .replaceAll(/\n\n+/g, '\n\n');
 
   document.content_title = document.content_title.split('|')[0].trim();
 
+  // specific to security: remove rule query section as it's usually large without much value for the LLM
+  if (document.product_name === 'security') {
+    const ruleQueryTitle = '### Rule query';
+    const ruleQueryPos = document.content_body.indexOf(ruleQueryTitle);
+    if (ruleQueryPos > -1) {
+      document.content_body = document.content_body.substring(0, ruleQueryPos);
```

Comment:
`### Rule query` is always the last part of these documents?

Reply:
Yeah, it's the last section of their rule page template